desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -57,43 +59,91 @@ def slack_events(request):
       parse_events(event_message)
   except ValueError as e:
     raise PopupException(_("Response content is not valid JSON"), detail=e)
-  
+
   return HttpResponse(status=200)
 
 
-def parse_events(event_message):
-  user_id = event_message.get('user')
-  text = event_message.get('text')
-  channel = event_message.get('channel')
-
-  BOT_ID = None
-  if appname is not None:
-    BOT_ID = get_bot_id(appname)
-
-  # ignore bot's own message
-  if BOT_ID == user_id:
-    return HttpResponse(status=200)
-  
-  if slack_client is not None:
-    if 'hello hue' in text.lower():
-      response = say_hi_user(channel, user_id)
-      if response['ok']:
-        return HttpResponse(status=200)
-      else:
-        raise PopupException(response["error"])
-
-  
-def say_hi_user(channel, user_id):
-  """Bot sends Hi<username> message in a specific channel"""
+def parse_events(event):
+  """
+  Parses the event according to its 'type'.
+  """
+
+  # To-do: Creating function for every type to de-congest
+  if event.get('type') == 'message':
+    user_id = event.get('user')
+    text = event.get('text')
+    channel = event.get('channel')
+
+    # ignore bot's own message since that will cause an infinite loop of messages if we respond.
+    if event.get('bot_id'):
+      return HttpResponse(status=200)
+    
+    if slack_client:
+      if text and 'hello hue' in text.lower():
+        response = say_hi_user(channel, user_id)
+        if response['ok']:
+          return HttpResponse(status=200)
+        else:
+          raise PopupException(response["error"])
 
+  if event.get('type') == 'link_shared':
```

Comment:
Move to its own `handle_shared_link_event` function? (think same ideas as `say_hi_user()`)

Reply:
And then please a unit test testing with links from query history, gist and invalid link :)